smoke-tests/spec/psp_spec.rb

```diff
@@ -0,0 +1,106 @@
+require "spec_helper"
+
+# Pod security policies (PSP) is a cluster-level resource that
+# controls security sensitive aspects of the pod specification.
+# This spec confirms the Cloud Platform psp's are operational
+# within its cluster.
+describe "pod security policies" do
+  let(:namespace) { "integrationtest-psp-#{readable_timestamp}" }
+  let(:pods) { get_running_pods(namespace)}
+
+  # Confirms the main psp's currently exist
+  specify "has the expected policies" do
+    names = get_psp.map { |set| set.dig("metadata", "name") }.sort
+
+    expected = [
+      "privileged",
+      "restricted",
+      "kube-system"
+    ]
+    expect(names).to include(*expected)
+  end
+
+  # Runs a privilege namespace and confirms both privilege and
+  # non-privilege containers can run. 
+  context "when namespace is privileged" do
+    before do
```

Comment:
I'd probably use something in here to explain that the erb template is part of what distinguishes privileged and non-privileged namespaces, e.g.
```
context "when namespace is privileged" do
  let(:deployment_file) { "spec/fixtures/privileged-deployment.yaml.erb" }
  ...
```
And similarly for unprivileged.

That would mean reorganising things so that it's the *context* which is privileged or not, but I think that would make the tests clearer and a little more concise.

Reply:
I originally used the organisation you outline but decided to change it to make as few function calls as possible. Reading your comment, I think you're right so I have decided to revert back.